plugins/auth-backend/src/providers/microsoft/provider.ts

```diff
@@ -205,32 +190,33 @@ export class MicrosoftAuthProvider implements OAuthHandlers {
   }
 }
 
-export const createMicrosoftProvider: AuthProviderFactory = ({
-  providerId,
-  globalConfig,
-  config,
-  tokenIssuer,
-}) =>
-  OAuthEnvironmentHandler.mapConfig(config, envConfig => {
-    const clientId = envConfig.getString('clientId');
-    const clientSecret = envConfig.getString('clientSecret');
-    const tenantID = envConfig.getString('tenantId');
-
-    const callbackUrl = `${globalConfig.baseUrl}/${providerId}/handler/frame`;
-    const authorizationUrl = `https://login.microsoftonline.com/${tenantID}/oauth2/v2.0/authorize`;
-    const tokenUrl = `https://login.microsoftonline.com/${tenantID}/oauth2/v2.0/token`;
-
-    const provider = new MicrosoftAuthProvider({
-      clientId,
-      clientSecret,
-      callbackUrl,
-      authorizationUrl,
-      tokenUrl,
+export type MicrosoftProviderOptions = {};
+
+export const createMicrosoftProvider = (
+  _options?: MicrosoftProviderOptions,
+): AuthProviderFactory => {
+  return ({ providerId, globalConfig, config, tokenIssuer }) =>
+    OAuthEnvironmentHandler.mapConfig(config, envConfig => {
+      const clientId = envConfig.getString('clientId');
+      const clientSecret = envConfig.getString('clientSecret');
+      const tenantID = envConfig.getString('tenantId');
```

Comment:
```suggestion
      const tenantId = envConfig.getString('tenantId');
```

Reply:
I mean sure, because I need to fix other stuff, but really not part of this PR xP